### PR TITLE
replxx: init at 0.0.4

### PR DESCRIPTION
**Pull request by ghost**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -34516,6 +34516,8 @@ with pkgs;
 
   rep-gtk = callPackage ../development/libraries/rep-gtk { };
 
+  replxx = callPackage ../development/libraries/replxx { };
+
```
Comment (Lurkki14) [inline pkgs/top-level/all-packages.nix:34520]: Placing packages in `pkgs/by-name` is the new recommended way for new packages: https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/re/replxx/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+, enableStatic ? stdenv.hostPlatform.isStatic
+}:
+
+stdenv.mkDerivation rec {
+  pname = "replxx";
+  version = "0.0.4";
+
+  src = fetchFromGitHub {
+    owner = "AmokHuginnsson";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/by-name/re/replxx/package.nix:14]:
```suggestion
    repo = "replxx";
```

File: pkgs/by-name/re/replxx/package.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, enableStatic ? stdenv.hostPlatform.isStatic
+}:
+
+stdenv.mkDerivation rec {
```
Comment (AndersonTorres) [inline pkgs/by-name/re/replxx/package.nix:8]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/re/replxx/package.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  pname = "replxx";
+  version = "0.0.4";
+
+  src = fetchFromGitHub {
+    owner = "AmokHuginnsson";
+    repo = pname;
+    rev = "release-${version}";
```
Comment (AndersonTorres) [inline pkgs/by-name/re/replxx/package.nix:15]:
```suggestion
    rev = "release-${finalAttrs.version}";
```

File: pkgs/by-name/re/replxx/package.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  cmakeFlags = [ "-DBUILD_SHARED_LIBS=${if enableStatic then "OFF" else "ON"}" ];
+
+  meta = with lib; {
+    homepage = "https://github.com/AmokHuginnsson/replxx";
+    description = "A readline and libedit replacement that supports UTF-8, syntax highlighting, hints and Windows and is BSD licensed";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ rs0vere ];
+    platforms = platforms.all;
+  };
+}
```
Comment (AndersonTorres) [inline pkgs/by-name/re/replxx/package.nix:30]:
```suggestion
})
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/re/replxx/package.nix (hunk 2 above)
Comment (rs0vere) [reply pkgs/by-name/re/replxx/package.nix]: I fixed, thanks

**Review by AndersonTorres — APPROVED** [review]
Ping me 36h to merge.

**Comment by rs0vere** [comment]
@AndersonTorres Pong!

**Pull request merged**